src/CarbonAware.Aggregators/test/CarbonAwareAggregatorTests.cs

```diff
@@ -202,4 +203,77 @@ public void TestGetCurrentForecastDataAsync_InvalidStartEndTimes_ThrowsException
         };
         Assert.ThrowsAsync<ArgumentException>(async () => await Aggregator.GetCurrentForecastDataAsync(props));
     }
+
+    [TestCase("eastus", "2021-11-18T00:00:00Z", "2021-11-18T08:00:00Z", ExpectedResult = 60)]
+    [TestCase("westus", "2021-11-17T00:00:00Z", "2021-11-18T00:00:00Z", ExpectedResult = 20)]
+    [TestCase("eastus", "2021-11-19T00:00:00Z", "2021-12-30T00:00:00Z", ExpectedResult = 0)]
+    [TestCase("fakelocation", "2021-11-18T00:00:00Z", "2021-12-30T00:00:00Z", ExpectedResult = 0)]
+    public async Task<double> CalculateAverageCarbonIntensityAsync_ValidTimeInterval(string regionName, string startString, string endString)
+    {
+        // Arrange
+        var location = new Location()
+        {
+            LocationType = LocationType.CloudProvider,
+            CloudProvider = CloudProvider.Azure,
+            RegionName = regionName
+        };
+
+        List<Location> locations = new List<Location>() {
+          location
+        };
+
+        DateTimeOffset start, end;
+        DateTimeOffset.TryParse(startString, out start);
+        DateTimeOffset.TryParse(endString, out end);
+
+        var props = new Dictionary<string, object?>()
+        {
+            { CarbonAwareConstants.Locations, locations },
+            { CarbonAwareConstants.Start, start },
+            { CarbonAwareConstants.End, end }
+        };
+
+        this.CarbonIntensityDataSource.Setup(x => x.GetCarbonIntensityAsync(It.IsAny<IEnumerable<Location>>(),
+            It.IsAny<DateTimeOffset>(), It.IsAny<DateTimeOffset>()))
+            .ReturnsAsync(TestData.GetFilteredEmissionDataList(location.RegionName, startString, endString));
+
+        // Act
+        var result = await this.Aggregator.CalculateAverageCarbonIntensityAsync(props);
+
+        // Assert
+        this.CarbonIntensityDataSource.Verify(r => r.GetCarbonIntensityAsync(locations, start, end), Times.Once);
+        return result;
+    }
+
+    [Test]
+    public async Task CalculateAverageCarbonIntensityAsync_UnderspecifiedTimeInterval()
+    {
+        // Arrange
+        var location = new Location()
+        {
+            LocationType = LocationType.Geoposition,
+            Latitude = (decimal)1.0,
```

Comment:
nit: You can use 
```suggestion
            Latitude = 1.0m,
```

Reply:
that's true -- for context, that test was "lift-n-shifted" from the other aggregator ...